test/system_test.py

```diff
@@ -293,7 +297,7 @@ def test_install_c2sm_master_nvhpc(self):
 class IconToolsTest(unittest.TestCase):
 
     def test_install(self):
-        spack_install_and_test('icontools @2.5.2')
+        spack_install_and_test('icontools @2.5.2%gcc')
```

Comment:
This adds unnecessary complexity, because figuring out a compatible compiler is the job of the concretizer.

Reply:
I added this because it picked cce on Daint.
But let me fix that on the level of sysconfig/daint